test/WebJobs.Script.Tests.Integration/TestScriptHostBuilder.cs

```diff
@@ -0,0 +1,65 @@
+﻿using System;
+using System.Linq;
+using Microsoft.Azure.WebJobs.Hosting;
+using Microsoft.Azure.WebJobs.Script.WebHost;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Options;
+
+namespace Microsoft.Azure.WebJobs.Script.Tests.Integration
+{
+    internal class TestScriptHostBuilder : IScriptHostBuilder
```

Comment:
This is a copy/paste of DefaultScriptHostBuilder with the configurable Action called before Build(). That class is sealed (by design?), or I'd add a virtual method and derive from it. But I'm not sure we want to make it easy to do that...